### PR TITLE
chore(connect): split types of different Connect implementations

### PR DESCRIPTION
**Pull request by martykan**
## Description

Previously, the settings and methods were the same for all Connect packages (eg. connect, connect-web, connect-webextension, connect-mobile).
This is not ideal, because some methods and settings are only available in some of them. 

The goal is to make it possible for each package to have it's own type - specifically for the settings and for additional methods besides the common ones.

### CODE REVIEW

File: packages/connect-web/src/impl/core-in-iframe.ts [hunk 1]
```diff
@@ -221,6 +227,7 @@ export class CoreInIframe implements ConnectFactoryDependencies {
 
             // auto init with default settings
             try {
+                // @ts-expect-error manifest could have been changed in the meantime, hard to check this
```
Comment (martykan) [inline packages/connect-web/src/impl/core-in-iframe.ts:230]: previously this worked due to weaker typing

File: packages/connect/src/types/settings.ts [hunk 2]
```diff
@@ -13,27 +13,20 @@ export interface ConnectSettingsPublic {
     manifest?: Manifest;
     connectSrc?: string;
     debug?: boolean;
-    hostLabel?: string;
-    hostIcon?: string;
     popup?: boolean;
     transportReconnect?: boolean;
-    webusb?: boolean; // deprecated
+    /** @deprecated */
+    webusb?: boolean;
```
Comment (mroz22) [inline packages/connect/src/types/settings.ts]: are we ready to remove it from types? (backwards comp stuff)?

**Review by mroz22 — APPROVED** [review]
I think it makes sense

File: packages/connect/src/types/settings.ts (hunk 2 above)
Comment (martykan) [reply packages/connect/src/types/settings.ts]: We can remove it, it looks like it doesn't even work anymore

**Pull request merged**